### PR TITLE
🏗 Clean up / replace lazy `require` statements in `build-system/`

### PR DESCRIPTION
**Pull request by rsimha**
**PR Highlights:**
- Remove the unused `compileTs()`, delete `compile/typescript.js`, and uninstall `tsickle` (https://github.com/ampproject/amphtml/pull/33376#discussion_r597352055)
- Use dynamic `import` statements where possible (https://github.com/ampproject/amphtml/pull/33376#pullrequestreview-616519500)
- Move `request.post` initialization code to `requestPost()`  (https://github.com/ampproject/amphtml/pull/33376#discussion_r597766552)
- Bonus fix: Minor cosmetic fixes to startup / error logging in `amp` task runner

Follow up to #33376

### CODE REVIEW

File: build-system/tasks/bundle-size/index.js [hunk 1]
```diff
@@ -115,8 +124,7 @@ async function storeBundleSize() {
   const commitHash = gitCommitHash();
   log('Storing bundle sizes for commit', cyan(shortSha(commitHash)) + '...');
   try {
-    const requestPost = util.promisify(require('request').post); // Lazy-required to speed up task loading.
-    const response = await requestPost({
+    const response = await getRequestPost()({
```
Comment (jridgewell) [inline build-system/tasks/bundle-size/index.js:127]: Pretty sure this requires a double `await`

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: build-system/tasks/bundle-size/index.js (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/bundle-size/index.js:127]: Yep. I did plan on checking the CI logs before merging this :)

image

**Review by kristoferbaxter — APPROVED** [review]
Looks great! Did this retain the expected performance boost from the CJS version?

**Comment by rsimha** [comment]
> Looks great! Did this retain the expected performance boost from the CJS version?

I believe so. Here's the median running time. (Although there is some variance. Probably because my 4 year old macbook is running a lot of stuff?)

image

File: build-system/tasks/bundle-size/index.js (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/bundle-size/index.js]: Fixed:
image

File: build-system/tasks/bundle-size/index.js [hunk 2]
```diff
@@ -115,7 +124,7 @@ async function storeBundleSize() {
   const commitHash = gitCommitHash();
   log('Storing bundle sizes for commit', cyan(shortSha(commitHash)) + '...');
   try {
-    const requestPost = util.promisify(require('request').post); // Lazy-required to speed up task loading.
+    const requestPost = await getRequestPost();
     const response = await requestPost({
```
Comment (jridgewell) [inline build-system/tasks/bundle-size/index.js:128]: Nit: You could pass the params to `getRequestPost` (and rename to `requestPost`):

```js
async function requestPost(...args) {
  const {default: request} = await import('request');
  return util.promisify(request.post)(...args);
}

**Review by jridgewell — APPROVED** [review]

File: build-system/tasks/bundle-size/index.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/bundle-size/index.js:129]: Final version worked:

image

**Pull request merged**